app.py

```diff
@@ -395,43 +395,56 @@ def _get_risk_level(debt_to_supply_ratio: float) -> str:
     with col1:
         user = streamlit.text_input("User")
         protocol = streamlit.text_input("Protocol")
+        
         users_and_protocols_with_debt = list(
             loans_data.loc[
                 loans_data['Debt (USD)'] > 0,
                 ['User', 'Protocol'],
-            ].itertuples(index = False, name = None)
+            ].itertuples(index=False, name=None)
         )
+    
         random_user, random_protocol = users_and_protocols_with_debt[numpy.random.randint(len(users_and_protocols_with_debt))]
+        
         if not user:
             streamlit.write(f'Selected random user = {random_user}.')
             user = random_user
         if not protocol:
             streamlit.write(f'Selected random protocol = {random_protocol}.')
             protocol = random_protocol
+
     loan = loans_data.loc[
         (loans_data['User'] == user)
         & (loans_data['Protocol'] == protocol),
     ]
-    collateral_usd_amounts, debt_usd_amounts = src.main_chart.get_specific_loan_usd_amounts(loan = loan)
-    with col2:
-        figure = plotly.express.pie(
-            collateral_usd_amounts,
-            values='amount_usd',
-            names='token',
-            title='Collateral (USD)',
-            color_discrete_sequence=plotly.express.colors.sequential.Oranges_r,
-        )
-        streamlit.plotly_chart(figure, True)
-    with col3:
-        figure = plotly.express.pie(
-            debt_usd_amounts,
-            values='amount_usd',
-            names='token',
-            title='Debt (USD)',
-            color_discrete_sequence=plotly.express.colors.sequential.Greens_r,
-        )
-        streamlit.plotly_chart(figure, True)
-    streamlit.dataframe(loan)
+
+    if loan.empty:
+        streamlit.warning(f"No loan found for user = {user} and protocol = {protocol}.")
+    else:
+        try:
+            collateral_usd_amounts, debt_usd_amounts = src.main_chart.get_specific_loan_usd_amounts(loan=loan)
+        except Exception as e:
```

Comment:
Pls be more specific about which Exception are you catching. I haven't seen the code failing here except for when it's missing, which is already handled (thanks to you 🙂).